src/main/java/nextstep/courses/domain/ProvidePolicy.java

```diff
@@ -0,0 +1,68 @@
+package nextstep.courses.domain;
+
+import nextstep.courses.CanNotCreateException;
+import nextstep.courses.CanNotJoinException;
+import nextstep.courses.enumerate.ProvideType;
+import nextstep.payments.domain.Payment;
+
+public class ProvidePolicy {
+    
+    private final Integer maxEnrollment;
+    private final Long tuitionFee;
+    
+    public ProvidePolicy() throws CanNotCreateException {
+        this(null, null);
+    }
+    
+    public ProvidePolicy(Integer maxEnrollment, Long tuitionFee) throws CanNotCreateException {
+        validate(maxEnrollment, tuitionFee);
+        this.maxEnrollment = maxEnrollment;
+        this.tuitionFee = tuitionFee;
+    }
+    
+    private void validate(Integer maxEnrollment, Long tuitionFee) throws CanNotCreateException {
+        if(IsMaxEnrollmentOnlyNull(maxEnrollment, tuitionFee)) {
+            throw new CanNotCreateException("하나만 무료정책을 가질 수없다 (하나만 null 일 수 없다)");
+        }
+        if(isTuitionFeeOnlyNull(maxEnrollment, tuitionFee)) {
+            throw new CanNotCreateException("하나만 무료정책을 가질 수없다 (하나만 null 일 수 없다)");
+        }
+    }
+    
+    private static boolean isTuitionFeeOnlyNull(Integer maxEnrollment, Long tuitionFee) {
+        return maxEnrollment != null && tuitionFee == null;
+    }
+    
+    private static boolean IsMaxEnrollmentOnlyNull(Integer maxEnrollment, Long tuitionFee) {
+        return maxEnrollment == null && tuitionFee != null;
+    }
+    
+    public void isCorrectPay(Payment payment) throws CanNotJoinException {
+        if(this.tuitionFee == null) {
+            throw new CanNotJoinException("무료 강의는 지불할 수 없다");
+        }
+        if(payment.isNotSameAmount(this.tuitionFee)) {
+            throw new CanNotJoinException("지불한 금액과 수강료 금액이 다르다");
+        }
+    }
+    
+    public void isAvailableEnroll(EnrolledUsers enrolledUsers) throws CanNotJoinException {
```

Comment:
```suggestion
    public void enroll(EnrolledUsers enrolledUsers) throws CanNotJoinException {
```
보통 is로 시작하는 메서드는 반환값이 boolean을 것을 기대함
AI를 통해 더 적절한 이름을 추천 받아보면 어떨까?